### PR TITLE
Remove metav1.TypeMeta field and implementation of runtime.Object interface from visibility.PendingWorkload, 

### PR DESCRIPTION
**Pull request by PBundyra**
#### What type of PR is this?
/kind cleanup
/kind api-change

#### What this PR does / why we need it:
Visibility.PendingWorkload does not need to implement runtime.Object nor have the field metav1.TypeMeta

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Visibility.PendingWorkload does not implement runtime.Object interface anymore

```

### CODE REVIEW

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

File: apis/visibility/v1alpha1/types.go [hunk 1]
```diff
@@ -80,6 +59,24 @@ type LocalQueueList struct {
 	Items []LocalQueue `json:"items"`
 }
 
+// PendingWorkload is a user-facing representation of a pending workload that summarizes the relevant information for
```
Comment (mimowo) [inline apis/visibility/v1alpha1/types.go:62]: Why moving this declaration? It increases the diff unnecessarily.

**Comment by alculquicondor** [comment]
/release-note-edit
```release-note
NONE
```

**Pull request merged**